dashboard/src/main/java/com/google/cloud/tools/opensource/cloudbomdashboard/VersionData.java

```diff
@@ -0,0 +1,277 @@
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.cloudbomdashboard;
+
+import java.io.BufferedInputStream;
+import java.io.FileOutputStream;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.HashSet;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.commons.io.FileUtils;
+import org.eclipse.aether.artifact.Artifact;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.Date;
+
+/**
+ * Container class for the conversion of a dashboard into a template
+ */
+public class VersionData {
+
+  /* There can only be one version of our 'All Versions' page */
+  public static final String ALL_VERSIONS_NAME = "all-versions";
+  private static final VersionData ALL_VERSIONS_DATA = new VersionData();
+
+  /* Helps to improve performance. No need to repeatedly look up remote resources. */
+  private static final Map<String, String> pomToDependenciesVersion = new HashMap<>();
+  private static final Map<Artifact, String> artifactToTime = new HashMap<>();
+  private static final Map<Artifact, String> artifactToLatestVersion = new HashMap<>();
+
+  private final Set<String> versions = new HashSet<>();
+
+  /* Everything associated with the template for this VersionData */
+  private Set<Artifact> artifacts = new HashSet<>();
+  private Set<String> artifactIds = new HashSet<>();
+
+  private static DateFormat dateFormat = new SimpleDateFormat("yyyyMMddhhmmss");
+  private static DateFormat outputFormat = new SimpleDateFormat("MM-dd-yyyy");
+
+  public VersionData(String cloudBomVersion) {
+    versions.add(cloudBomVersion);
+  }
+
+  private VersionData() {
+  }
+
+  public void addData(Set<Artifact> artifacts) {
+    this.artifacts.addAll(artifacts);
+  }
+
+  public static void addArtifactsToAllVersions(Set<Artifact> artifacts) {
+    ALL_VERSIONS_DATA.addData(artifacts);
+  }
+
+  public static void addVersionToAllVersions(String version) {
+    ALL_VERSIONS_DATA.versions.add(version);
+  }
+
+  /**
+   * Creates new mapping of template data for use in creating the All Versions page.
+   */
+  public static Map<String, Object> getAllVersionsTemplate() {
+    return ALL_VERSIONS_DATA.getTemplateData();
+  }
+
+  /**
+   * Returns the 'template data' formatting of our data.
+   */
+  public Map<String, Object> getTemplateData() {
+    //Freemarker template
+    Map<String, Object> templateData = new HashMap<>();
+
+    //Mappings used within Freemarker file
+    Map<String, String> currentVersion = new HashMap<>();
```

Comment:
e.g. here I don't know whether the currentVersion is the key or the value, and I have no idea what the other string is.